components/utilities/combination.js

```diff
@@ -0,0 +1,176 @@
+// Function validateCombo that takes in an array of cards which is the current players choice of cards, and matches it to the current cycles combination. 
+export const dictionaryCombinations = {
+  'single': {
+    // Can't be more than 1 card.
+    isValid: (cards) => cards.length === 1,
+  },
+  'pair': {
+    // Comprised of two cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 2) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number;
+    },
+  },
+  'triplet': {
+    // Comprised of three cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 3) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number && cards[0].number === cards[2].number;
+    }
+  },
+  'quartet': {
+    // Comprised of four cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 4) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number && cards[0].number === cards[2].number && cards[0].number === cards[3].number;
+    }
+  },
+  'sequence': {
+    // Two important conditions
+    // 1: length of 3 or more numbers
+    // 2: cards in order.
+    isValid: (cards) => {
+
+      if (cards.length < 3) {
+        return false;
+      }
+    
+      let containsDupe = false;
+      const uniqNumbers = {};
+      for (let i = 0; i < cards.length; i++) {
+        if (uniqNumbers[cards[i].number] === undefined) {
+          uniqNumbers[cards[i].number] = true;
+        } else {
+          containsDupe = true;
+        }
```

Comment:
can consolidate the logic using a reduce like so (this is untested, so pls test):

```javascript
const {uniqNumbers, containsDupe} = cards.reduce((acc, {number}) => {
  if(acc.uniqNumbers.includes(number)) {
    acc.containsDupe = true;
  } else {
    acc.uniqNumbers.push(number);
  }
  return acc;
}, {uniqNumbers: [], containsDupe: false});
```

this will give you the same boolean value for `containsDupe` as well as an array of the numbers so you don't need to re-iterate through the cards array and create an array from an object like you are below.

Reply:
This is great refactoring, I used this template for double sequence as well.